master/internal/proxy/proxy.go

```diff
@@ -156,14 +163,59 @@ func (p *Proxy) newProxyHandler(serviceID string) echo.HandlerFunc {
 		case c.IsWebSocket():
 			proxy = newSingleHostReverseWebSocketProxy(c, service.URL)
 		default:
-			proxy = httputil.NewSingleHostReverseProxy(service.URL)
+			newProxy, err := setupProxy(service.URL)
+			if err != nil {
+				return err
+			}
+
+			proxy = newProxy
 		}
 		proxy.ServeHTTP(c.Response(), req)
 
 		return nil
 	}
 }
 
+func setupProxy(serviceURL *url.URL) (*httputil.ReverseProxy, error) {
```

Comment:
Nit: I think this should be `setUpProxy` since "setup" is a noun and this is the verb form, which is two words: "set up [a] proxy."